core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SPARQLProtocolSession.java

```diff
@@ -148,7 +148,7 @@
 	 */
 	private final int maximumUrlLength;
 
-	final Logger logger = LoggerFactory.getLogger(this.getClass());
+	final Logger logger = LoggerFactory.getLogger(SPARQLProtocolSession.class);
```

Comment:
Btw. I prefer final static and possibly private for loggers.

Reply:
I agree with preferring private: as this example shows, inheriting loggers is just a recipe for confusion. Unfortunately, this was designed for this purpose in the past, so we're stuck with it at least until the next major release.

I'm not sure why you prefer static though. We use a LoggerFactory so there is no overhead - SLF4J caches instances per class name, so it's a singleton. And there may be issues caused by static logger references in some applications (see https://cwiki.apache.org/confluence/display/COMMONS/Logging+StaticLog ).

Reply:
I wasn't aware of that issue. I mostly like static loggers because you can reference them from static context too.